packages/xod-project/src/patch.js

```diff
@@ -887,3 +887,123 @@ export const removeDebugNodes = def(
     listNodes,
   )(patch)
 );
+
+/**
+ * Returns a map of pins for patch that exists.
+ * So Pins are fully valid, contains proper types and etc.
+ */
+export const getPinsForExistentPatch = def(
+  'getPinsForExistentPatch :: Node -> Patch -> Map PinKey Pin',
+  (node, patch) => R.compose(
```

Comment:
As far as I understand, we’re getting pins for a node, not for a patch. So the name is misguiding. IMO better: `listExistingNodePins`

Given that, the function is better suitable for `node.js`. In that case, it hasn't even to be exported.

Also, BTW, “existent” is awkward, “existing“ is much more casual.